src/3d/Ast.fst

```diff
@@ -286,12 +286,18 @@ type array_qualifier =
   | ArrayByteSizeAtMost //[:byte-size-at-most
   | ArrayByteSizeSingleElementArray //[:byte-size-single-element-array
 
+noeq
+type field_array_t =
+  | FieldArrayNormal

```

Comment:
this case indicates that the field is not an array, right?

Reply:
yes, the field is neither a string nor otherwise an array